expression/builtin_time_vec.go

```diff
@@ -1989,11 +1987,93 @@ func (b *builtinCurrentDateSig) vecEvalTime(input *chunk.Chunk, result *chunk.Co
 }
 
 func (b *builtinMakeTimeSig) vectorized() bool {
-	return false
+	return true
+}
+
+func (b *builtinMakeTimeSig) getVecIntParam(arg Expression, input *chunk.Chunk, col *chunk.Column) (err error) {
+	if arg.GetType().EvalType() == types.ETReal {
+		err = arg.VecEvalReal(b.ctx, input, col)
+		if err != nil {
+			return err
+		}
+		f64s := col.Float64s()
+		i64s := col.Int64s()
+		n := input.NumRows()
+		for i := 0; i < n; i++ {
+			i64s[i] = int64(f64s[i])
+		}
+		return nil
+	}
+	err = arg.VecEvalInt(b.ctx, input, col)
+	return err
 }
 
 func (b *builtinMakeTimeSig) vecEvalDuration(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	result.ResizeInt64(n, false)
+	hoursBuf := result
+	var err error
+	if err = b.getVecIntParam(b.args[0], input, hoursBuf); err != nil {
+		return err
+	}
+	minutesBuf, err := b.bufAllocator.get(types.ETInt, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(minutesBuf)
+	if err = b.getVecIntParam(b.args[1], input, minutesBuf); err != nil {
+		return err
+	}
+	secondsBuf, err := b.bufAllocator.get(types.ETReal, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(secondsBuf)
+	if err = b.args[2].VecEvalReal(b.ctx, input, secondsBuf); err != nil {
+		return err
+	}
+	hours := hoursBuf.Int64s()
+	minutes := minutesBuf.Int64s()
+	seconds := secondsBuf.Float64s()
+	durs := result.GoDurations()
+	result.MergeNulls(minutesBuf, secondsBuf)
+	for i := 0; i < n; i++ {
+		if result.IsNull(i) {
+			continue
+		}
+		if minutes[i] < 0 || minutes[i] >= 60 || seconds[i] < 0 || seconds[i] >= 60 {
+			result.SetNull(i, true)
+			continue
+		}
+		overflow := false
+		if hours[i] < 0 && mysql.HasUnsignedFlag(b.args[0].GetType().Flag) {
```

Comment:
We can cache the result of `mysql.HasUnsignedFlag(b.args[0].GetType().Flag)` outside the for loop.

Reply:
We should add test for for `UnsignedFlag`